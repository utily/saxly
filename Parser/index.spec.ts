import "jest"
import { saxly } from "../index"

describe("saxly.Parser", () => {
	it("parse <root/>", async () => expect(await saxly.Parser.parse("<root/>")).toMatchSnapshot())
	it("parse <root></root>", async () => expect(await saxly.Parser.parse("<root></root>")).toMatchSnapshot())
	it('parse <root attribute="value"></root>', async () =>
		expect(await saxly.Parser.parse('<root attribute="value"></root>')).toMatchSnapshot())
	it("parse <root readonly></root>", async () =>
		expect(await saxly.Parser.parse("<root readonly></root>")).toMatchSnapshot())
	it("parse <root>text</root>", async () => expect(await saxly.Parser.parse("<root>text</root>")).toMatchSnapshot())
	it("parse <root>first name</root>", async () =>
		expect(await saxly.Parser.parse("<root>first name</root>")).toMatchSnapshot())
	it('parse <?xml version="1.0" language="en/US"?>\n<root/>', async () =>
		expect(await saxly.Parser.parse('<?xml version="1.0" language="en/US"?>\n<root/>')).toMatchSnapshot())
	it("parse <root><nested/></root>", async () =>
		expect(await saxly.Parser.parse("<root><nested/></root>")).toMatchSnapshot())
	it("parse <root><nested></nested></root>", async () =>
		expect(await saxly.Parser.parse("<root><nested></nested></root>")).toMatchSnapshot())
	it("parse <root>first<nested></nested>last</root>", async () =>
		expect(await saxly.Parser.parse("<root>first<nested></nested>last</root>")).toMatchSnapshot())
	it("parse <root><nested><inner/></nested></root>", async () =>
		expect(await saxly.Parser.parse("<root><nested><inner/></nested></root>")).toMatchSnapshot())
	it("parse <root>first<nested>middle</nested>last</root>", async () =>
		expect(await saxly.Parser.parse("<root>first<nested>middle</nested>last</root>")).toMatchSnapshot())
	it('parse <root attribute="value">\nText<element key="value" readonly>last name</element></root>', async () =>
		expect(
			await saxly.Parser.parse('<root attribute="value">\nText<element key="value" readonly>last name</element></root>')
		).toMatchSnapshot())
})
