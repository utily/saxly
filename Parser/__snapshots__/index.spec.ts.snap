// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saxly.Parser parse <?xml version="1.0" language="en/US"?>
<root/> 1`] = `
{
  "declaration": {
    "attributes": {
      "language": "en/US",
      "version": "1.0",
    },
    "type": "xml",
  },
  "errors": [],
  "root": {
    "attributes": {},
    "content": [],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root attribute="value">
Text<element key="value" readonly>last name</element></root> 1`] = `
{
  "declaration": undefined,
  "errors": [
    {
      "error": "unknown",
      "range": {
        "end": {
          "column": 0,
          "row": 1,
        },
        "start": {
          "column": 9,
          "row": 0,
        },
      },
    },
    {
      "error": "expected start element end",
      "range": {
        "end": {
          "column": 9,
          "row": 1,
        },
        "start": {
          "column": 8,
          "row": 1,
        },
      },
    },
    {
      "error": "unknown",
      "range": {
        "end": {
          "column": 10,
          "row": 1,
        },
        "start": {
          "column": 9,
          "row": 1,
        },
      },
    },
    {
      "error": "unknown",
      "range": {
        "end": {
          "column": 11,
          "row": 1,
        },
        "start": {
          "column": 10,
          "row": 1,
        },
      },
    },
    {
      "error": "unknown",
      "range": {
        "end": {
          "column": 12,
          "row": 1,
        },
        "start": {
          "column": 11,
          "row": 1,
        },
      },
    },
    {
      "error": "unknown",
      "range": {
        "end": {
          "column": 14,
          "row": 1,
        },
        "start": {
          "column": 13,
          "row": 1,
        },
      },
    },
  ],
  "root": {
    "attributes": {
      "attribute": "value",
    },
    "content": [
      {
        "attributes": {
          "key": "value",
        },
        "content": [],
        "name": "element",
      },
    ],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root attribute="value"></root> 1`] = `
{
  "declaration": undefined,
  "errors": [],
  "root": {
    "attributes": {
      "attribute": "value",
    },
    "content": [],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root readonly></root> 1`] = `
{
  "declaration": undefined,
  "errors": [],
  "root": {
    "attributes": {
      "readonly": undefined,
    },
    "content": [],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root/> 1`] = `
{
  "declaration": undefined,
  "errors": [],
  "root": {
    "attributes": {},
    "content": [],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root></root> 1`] = `
{
  "declaration": undefined,
  "errors": [],
  "root": {
    "attributes": {},
    "content": [],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root><nested/></root> 1`] = `
{
  "declaration": undefined,
  "errors": [],
  "root": {
    "attributes": {},
    "content": [
      {
        "attributes": {},
        "content": [],
        "name": "nested",
      },
    ],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root><nested></nested></root> 1`] = `
{
  "declaration": undefined,
  "errors": [],
  "root": {
    "attributes": {},
    "content": [
      {
        "attributes": {},
        "content": [],
        "name": "nested",
      },
    ],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root>first name</root> 1`] = `
{
  "declaration": undefined,
  "errors": [],
  "root": {
    "attributes": {},
    "content": [
      "name",
      "first",
    ],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root>first<nested></nested>last</root> 1`] = `
{
  "declaration": undefined,
  "errors": [],
  "root": {
    "attributes": {},
    "content": [
      "last",
      {
        "attributes": {},
        "content": [],
        "name": "nested",
      },
      "first",
    ],
    "name": "root",
  },
}
`;

exports[`saxly.Parser parse <root>first<nested>middle</nested>last</root> 1`] = `
{
  "declaration": undefined,
  "errors": [
    {
      "error": "expected start element",
      "node": {
        "attributes": {},
        "content": [
          "middle",
        ],
        "name": "nested",
      },
    },
  ],
  "root": {
    "attributes": {},
    "content": [
      "last",
    ],
    "name": "nested",
  },
}
`;

exports[`saxly.Parser parse <root>text</root> 1`] = `
{
  "declaration": undefined,
  "errors": [],
  "root": {
    "attributes": {},
    "content": [
      "text",
    ],
    "name": "root",
  },
}
`;
